src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    public void getStatistic(String fromFile, String toFile) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFile))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                if (parts.length != 2) {
+                    continue;
+                }
+                String operation = parts[0].trim();
+                int amount = Integer.parseInt(parts[1].trim());
+
+                if (operation.equals("supply")) {
+                    supplyTotal += amount;
+                } else if (operation.equals("buy")) {
+                    buyTotal += amount;
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("Error reading file" + e.getMessage());
+        }
+        int result = supplyTotal - buyTotal;
 
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFile))) {
+            writer.write("supply," + supplyTotal);
+            writer.newLine();
+            writer.write("buy," + buyTotal);
+            writer.newLine();
+            writer.write("result," + result);
+        } catch (IOException e) {
+            System.out.println("Error writing file" + e.getMessage());
+        }
```

The catch block for writing prints an error instead of rethrowing a contextual RuntimeException. This also violates: "Exceptions must not be ignored; do not leave empty catch blocks or only print stack traces. Re-throw a RuntimeException with a meaningful message including the file name and the original exception as the cause [CHECKLIST ITEM #12]." Replace the `System.out.println` with `throw new RuntimeException("Can't write data to file " + toFile, e);` so callers/tests can observe the failure and you preserve the original exception as the cause. See the write catch block here .